### PR TITLE
Login

### PR DESCRIPTION
**Pull request by kaozdl**
add login and registration endpoints

### CODE REVIEW

File: users/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Generated by Django 2.1.5 on 2019-01-07 18:17
+
+import django.contrib.auth.models
+import django.contrib.auth.validators
+from django.db import migrations, models
```
Comment (GuilleLeopold) [inline users/migrations/0001_initial.py]: Shouldn't we sort this in some order? For example first the ones that are only import and then the ones that is specified the from

File: users/apps.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
```
Comment (GuilleLeopold) [inline users/apps.py]: I think you can remove two blank lines here, right?

File: users/migrations/0001_initial.py (hunk 1 above)
Comment (GuilleLeopold) [reply users/migrations/0001_initial.py]: Sorry, I just realized that this file was probably generated by make migrations

File: targets/models.py [hunk 3]
```diff
@@ -7,15 +7,15 @@ class Target(models.Model):
 
     #location define it later
     topic = models.CharField(max_length=20)
-    user = models.ForeignKey(settings.AUTH_USER_MODEL)
+    user = models.ForeignKey(settings.AUTH_USER_MODEL,on_delete=models.PROTECT)
```
Comment (GuilleLeopold) [inline targets/models.py]: Is it a good practice to leave a blank line after method firm?

File: targets/models.py [hunk 4]
```diff
@@ -7,15 +7,15 @@ class Target(models.Model):
...
 
 class Chat(models.Model):
     
-    user1 = models.ForeignKey(settings.AUTH_USER_MODEL)
-    user2 = models.ForeignKey(settings.AUTH_USER_MODEL)
+    user1 = models.ForeignKey(settings.AUTH_USER_MODEL,on_delete=models.PROTECT,related_name='user1_set')
```
Comment (GuilleLeopold) [inline targets/models.py]: missing space after commas in method attributes

File: users/apps.py (hunk 2 above)
Comment (kaozdl) [reply users/apps.py]: yes, the blank lines will be removed.

File: targets/models.py (hunk 4 above)
Comment (kaozdl) [reply targets/models.py]: This and the upper issue will be fixed in the next commit.

**Comment by kaozdl** [comment]
apps should be self contained with all the models they need to work.
Targets for instance has the Target, Chat and Message models in the models.py
and it should work fine with those even if we change the users app.

**Pull request merged**